internal/pkg/cli/pipeline_update.go

```diff
@@ -118,6 +126,51 @@ func (opts *UpdatePipelineOpts) getArtifactBuckets() ([]deploy.ArtifactBucket, e
 	return buckets, nil
 }
 
+func (opts *UpdatePipelineOpts) shouldUpdate() (bool, error) {
+	if opts.SkipConfirmation {
+		return true, nil
+	}
+
+	shouldUpdate, err := opts.prompt.Confirm(fmt.Sprintf(fmtUpdateEnvPrompt, opts.PipelineName), "")
+	if err != nil {
+		return false, fmt.Errorf("prompt for pipeline update: %w", err)
+	}
+	return shouldUpdate, nil
+}
+
+func (opts *UpdatePipelineOpts) deployPipeline(in *deploy.CreatePipelineInput) error {
+	exist, err := opts.pipelineDeployer.PipelineExist(in)
+	if err != nil {
+		return fmt.Errorf("check if pipeline exists: %w", err)
+	}
+	if !exist {
+		opts.prog.Start(fmt.Sprintf(fmtCreatePipelineStart, color.HighlightUserInput(opts.PipelineName)))
+		err := opts.pipelineDeployer.CreatePipeline(in)
+		var alreadyExists *cloudformation.ErrStackAlreadyExists
+		if err == nil || errors.As(err, &alreadyExists) {
+			opts.prog.Stop(log.Ssuccessf(fmtCreatePipelineComplete, color.HighlightUserInput(opts.PipelineName)))
+			return nil
+		}
+		opts.prog.Stop(log.Serrorf(fmtCreatePipelineFailed, color.HighlightUserInput(opts.PipelineName)))
+		return fmt.Errorf("create pipeline: %w", err)
```

Comment:
nit: What do you think about moving the failure case to the top?
```go
if err := opts.pipelineDeployer.CreatePipeline(in); err != nil {
  var alreadyExists *cloudformation.ErrStackAlreadyExists
  if !errors.As(err, &alreadyExists) {
     opts.prog.Stop(log.Serrorf(fmtCreatePipelineFailed, color.HighlightUserInput(opts.PipelineName)))
     return fmt.Errorf("create pipeline: %w", err)
  }
}
// success case
```

Reply:
That's much better :+1: